### PR TITLE
Add docs for environment variable subtitution

### PR DESCRIPTION
**Pull request by jbeisen**

### CODE REVIEW

**Review by mwylde — APPROVED** [review]

File: connectors/overview.mdx [hunk 1]
```diff
@@ -139,3 +139,27 @@ able to read data.
 
 To address this, source tasks with no work assigned to them will mark themselves as idle
 immediately.
+
+## Environment variable substitution
+
+Some fields in the connection configurations can use environment variables to substitute values
```
Comment (jacksonrnewhouse) [inline connectors/overview.mdx:145]: Is there a way we can document which fields support variable substitution?

Comment (jbeisen) [reply connectors/overview.mdx:145]: In the UI all fields that support substitution automatically say so in the description - 
image

Think we should mention here that it's controlled by the `format` attribute in the json schema?

Comment (jacksonrnewhouse) [reply connectors/overview.mdx:145]: Might be fine to just list them for the moment.

Comment (jbeisen) [reply connectors/overview.mdx:145]: I've added the line
> Currently, environment variable substitution is supported for fields containing API keys,
usernames, passwords, and HTTP headers.

**Pull request merged**